### PR TITLE
Added ways to cancel tealprints

### LINKED ISSUE #32: Need a way to cancel a tealprint
### Labels

[FEATURE]

### Description

Oh no, I put a tealprint down wrong, whatever shall I do?

### Steps to reproduce

1. Put down a tealprint in the wrong place
2. regret it

### Expected Behavior

I'm not sure how this should happen exactly, but some way of cancelling construction and dropping any resources that were stored within

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[QOL] [FEATURE]
## About the PR 
Fixes #32, afaik there was no way to cancel tealprints previously.

## Why's this needed? 
Drones wasting time building mis-click tealprints is bad

### CODE REVIEW

File: code/obj/flock/structure/flock_structure_ghost.dm [hunk 1]
```diff
@@ -46,3 +46,22 @@
...
+	playsound(src, 'sound/misc/flockmind/flockdrone_door_deny.ogg', 50, 1)
+	qdel(src)
+
+/obj/flock_structure/ghost/verb/cancel()
+	set src in view()
+	set category = "Local"
+	set name = "Cancel"
+	if (!istype(usr, /mob/living/intangible/flock/flockmind))
+		var/mob/living/critter/flock/drone/drone = usr
+		if (istype(drone) && istype(drone.controller, /mob/living/intangible/flock/flockmind))
```
Comment (amylizzle) [inline code/obj/flock/structure/flock_structure_ghost.dm:64]: I'm probably being pedantic here, but this should maybe be 
```
if(!istype(drone) || !istype(drone.controller, /mob/living/intangible/flock/flockmind)
    return
```

File: code/obj/flock/structure/flock_structure_ghost.dm [hunk 2]
```diff
@@ -46,3 +46,22 @@
...
+
+/obj/flock_structure/ghost/proc/cancelBuild()
+	if (currentmats > 0)
+		var/obj/item/flockcache/cache = new(get_turf(src))
+		cache.resources = currentmats
+	flock_speak(null, "Tealprint derealizing", flock)
+	playsound(src, 'sound/misc/flockmind/flockdrone_door_deny.ogg', 50, 1)
+	qdel(src)
+
+/obj/flock_structure/ghost/verb/cancel()
```
Comment (stonepillars) [inline code/obj/flock/structure/flock_structure_ghost.dm:58]: Verbs can be seen from anywhere and a bit cumbersome to use. Maybe just interacting with a ghost building can open some sort of UI or similar that has a button to cancel?

Comment (ZeWaka) [reply code/obj/flock/structure/flock_structure_ghost.dm:58]: Yeah, agree. We're looking to remove verbs entirely eventually.

Comment (ZeWaka) [reply code/obj/flock/structure/flock_structure_ghost.dm:58]: even just opening a tgui alert yes/no

Comment (TobleroneSwordfish) [reply code/obj/flock/structure/flock_structure_ghost.dm:58]: alright, I'll replace it with a tgui alert for now, and I'll probably add a cancel button to the flockpanel as well in future

File: code/obj/flock/structure/flock_structure_ghost.dm (hunk 1 above)
Comment (TobleroneSwordfish) [reply code/obj/flock/structure/flock_structure_ghost.dm]: I have no idea why I wrote it like that initially lol

**Comment by TobleroneSwordfish** [comment]
Upon changing the cancel option to an alert I've removed it from flockdrones since it's likely to get in the way as a drone. You can either eject or use the flock panel to cancel.

File: code/obj/flock/structure/flock_structure_ghost.dm [hunk 3]
```diff
@@ -28,6 +28,16 @@
 		flock_speak(null, "ERROR: No Structure Tealprint Assigned, Deleting", flock)
 		qdel(src) //no exist if building null
 
+/obj/flock_structure/ghost/attack_hand(mob/user)
+	. = ..()
+	boutput(user, "attackhand called")
```
Comment (amylizzle) [inline code/obj/flock/structure/flock_structure_ghost.dm:33]: debug code left in

**Review by amylizzle — CHANGES_REQUESTED** [review]

File: code/obj/flock/structure/flock_structure_ghost.dm (hunk 3 above)
Comment (TobleroneSwordfish) [reply code/obj/flock/structure/flock_structure_ghost.dm:33]: goddamnit

**Review by amylizzle — APPROVED** [review]

File: code/obj/flock/structure/flock_structure_ghost.dm [hunk 4]
```diff
@@ -28,6 +28,12 @@
 		flock_speak(null, "ERROR: No Structure Tealprint Assigned, Deleting", flock)
 		qdel(src) //no exist if building null
 
+/obj/flock_structure/ghost/Click(location, control, params)
+	if ("alt" in params2list(params) || !istype(usr, /mob/living/intangible/flock/flockmind))
```
Comment (stonepillars) [inline code/obj/flock/structure/flock_structure_ghost.dm:32]: Still appears if flockdrone controller is a flockmind, which can be annoying if the flockmind itself is trying to put stuff into a building

Comment (TobleroneSwordfish) [reply code/obj/flock/structure/flock_structure_ghost.dm]: TIL `in` binds like that, fixed.

**Pull request merged**